axon-framework/monitoring/tracing.md

```diff
@@ -0,0 +1,535 @@
+
+```java
+public class CustomSpanAttributesProvider implements SpanAttributesProvider {
+
+    @Nonnull
+    @Override
+    public Map<String, String> provideForMessage(@Nonnull Message<?> message) {
+        // Provide your labels based on the message here
+        return Collections.emptyMap();
+    }
+}
+```
+
+You can register this custom `SpanAttributesProvider` in one of the following ways.
+
+{% tabs %}
+{% tab title="Spring Boot Bean" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Bean
+    public SpanAttributesProvider customSpanAttributesProvider() {
+        // Autoconfiguration picks beans of type SpanAttributesProvider up automatically.
+        return new CustomSpanAttributesProvider();
+    }
+}
+```
+
+{% endtab %}
+{% tab title="Spring Boot injection" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Autowired
+    public void configureSpanFactory(SpanFactory spanFactory) {
+        spanFactory.registerSpanAttributeProvider(new CustomSpanAttributesProvider());
+    }
+}
+```
+
+{% endtab %}  
+{% tab title="Manual" %}
+
+```java
+public class AxonConfigurer {
+
+    public void configure(Configuration configuration) {
+        configuration.spanFactory().registerSpanAttributeProvider(new CustomSpanAttributesProvider());
+    }
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## OpenTelemetry <a id="opentelemetry"></a>
+
+Axon work provides [OpenTelemetry support](https://opentelemetry.io/docs/concepts/what-is-opentelemetry/) out of the
+box. The OpenTelemetry standard improves upon the OpenTracing standard by providing more
+auto-instrumentation without the need for the user to configure many things.
+
+OpenTelemetry works by adding a Java agent to the execution of the application. Based on the configuration, the agent
+will collect logs, metrics and tracing automatically before sending it to a collector that can provide insights.
+ElasticSearch APM, Jaeger and many other tools are available for this and the configuration of these is beyond the scope
+of this guide. You can find more
+information [in the "Getting Started" section of the OpenTelemetry documentation.](https://opentelemetry.io/docs/instrumentation/java/getting-started/)
+
+OpenTelemetry [supports a lot of libraries,
+frameworks and application servers out of the box.](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md)
+For example, when a Spring REST endpoint is called it will automatically start a trace.
+With the `axon-tracing-opentelemetry` module, this trace will be propagated to all subsequent Axon Framework messages.
+For example, if the REST call produces a command which is sent over Axon Server,
+handling the command will be included in the same trace as the original REST call.
+
+### Configuration
+
+To get OpenTelemetry support enabled you will need to add the following dependency to your application's dependencies:
+
+{% tabs %}
+{% tab title="Maven" %}
```

Comment:
This would only be sufficient with Spring Boot, right?
Without it, you will need to add the dependency *and* configure the `SpanFactory` to be an `OpenTelemetrySpanFactory`.

Reply:
Yes, that's in the next sentence: `Depending on your application, more configuration might be needed.`
There are two subsections, one for boot and one for custom